File: solutions-dotnet_fse/week-1/exercise4/exercise4.java
import java.util.*;

public class FinancialForecast {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        // Step 1: Input historical monthly income
        System.out.print("Enter the number of past months: ");
        int n = sc.nextInt();

        double[] income = new double[n];
        System.out.println("Enter income for each month:");
        for (int i = 0; i < n; i++) {
            System.out.print("Month " + (i + 1) + ": ₹");
            income[i] = sc.nextDouble();
        }

        // Step 2: Calculate average growth rate
        double totalGrowth = 0;
        for (int i = 1; i < n; i++) {
            totalGrowth += (income[i] - income[i - 1]) / income[i - 1];
        }

        double avgGrowthRate = totalGrowth / (n - 1);

        // Step 3: Forecast next month
        double forecast = income[n - 1] * (1 + avgGrowthRate);

        // Step 4: Output
        System.out.printf("\n Average Monthly Growth Rate: %.2f%%\n", avgGrowthRate * 100);
        System.out.printf("3 Forecasted Income for Next Month: ₹%.2f\n", forecast);

        sc.close();
    }
}

File: solutions-dotnet_fse/week-1/exercise4/exercise4.cs
class exercise4
{
    // Calculate Average Monthly Growth Rate
    static double CalculateAvgGrowthRate(double[] monthlyRevenue, int size)
    {
        double totalGrowth = 0.0;

        for (int i = 1; i < size; i++)
        {
            double growth = (monthlyRevenue[i] - monthlyRevenue[i - 1]) / monthlyRevenue[i - 1];
            totalGrowth += growth;
        }

        return totalGrowth / (size - 1);
    }

    // Forecast future revenue
    static double[] Forecast(double[] hist, int histSize, int n)
    {
        double[] forecast = new double[histSize + n];
        Array.Copy(hist, forecast, histSize);

        double avg = CalculateAvgGrowthRate(hist, histSize);
        double last = hist[histSize - 1];

        for (int i = histSize; i < histSize + n; i++)
        {
            last *= (1 + avg);
            forecast[i] = last;
        }

        return forecast;
    }

    static void Main()
    {
        Console.Write("Enter number of past months data: ");
        int n = int.Parse(Console.ReadLine() ?? "0");

        double[] past = new double[n];

        for (int i = 0; i < n; i++)
        {
            Console.Write($"Enter revenue for month {i + 1}: ");
            past[i] = double.Parse(Console.ReadLine() ?? "0");
        }

        Console.Write("Enter number of months to forecast: ");
        int m = int.Parse(Console.ReadLine() ?? "0");

        double[] result = Forecast(past, n, m);

        Console.WriteLine("\nHistorical Revenue Data");
        for (int i = 0; i < n; i++)
        {
            Console.WriteLine($"\tMonth {i + 1}: Rs.{result[i]:F2}");
        }

        Console.WriteLine("\nForecasted Revenue");
        for (int i = n; i < result.Length; i++)
        {
            Console.WriteLine($"\tMonth {i + 1}: Rs.{result[i]:F2}");
        }
    }
}
